cloudmarker/test/test_gcpcloud.py

```diff
@@ -145,6 +194,7 @@ def test_firewall_single_allowed_rule(self):
         }
         m = self._mock_discovery
         m.build().firewalls().list().execute.return_value = mock_firewall_dict
+        m.build().projects().list().execute.return_value = mock_projects_dict
```

Comment:
In each of these test methods, can you put the `m.build().projects()` statements before `m.build().firewalls()`? In places like this, I like to see the mocks being setup in the chronological order of usage. In the `GCPCloud` code, the projects would be retrieved before the firewalls, so I prefer maintaining the same order in the test code as well. This is not a strict rule. There are cases where test code neatness may trump chronological ordering as per usage.